lib/babe/build.go

```diff
@@ -222,22 +222,38 @@ func (b *BlockBuilder) buildBlockExtrinsics(slot Slot, rt runtime.Instance) []*t
 	return included
 }
 
-func buildBlockInherents(slot Slot, rt runtime.Instance) ([][]byte, error) {
+func buildBlockInherents(slot Slot, rt runtime.Instance, parent *types.Header) ([][]byte, error) {
 	// Setup inherents: add timstap0
 	idata := types.NewInherentsData()
 	timestamp := uint64(time.Now().UnixMilli())
-	err := idata.SetInt64Inherent(types.Timstap0, timestamp)
+	err := idata.SetInherent(types.Timstap0, timestamp)
 	if err != nil {
 		return nil, err
 	}
 
 	// add babeslot
-	err = idata.SetInt64Inherent(types.Babeslot, slot.number)
+	err = idata.SetInherent(types.Babeslot, slot.number)
 	if err != nil {
 		return nil, err
 	}
 
-	ienc, err := idata.Encode()
+	parachainInherent := ParachainInherentData{
```

Comment:
> I wonder if runtimes that don't require these inherents will just ignore this.

Yes, but is that problematic?

> shouldn't we only add the parachain inherent data for the polkadot/kusama/westend runtimes?

 It seems like currently polkadot uses `9281`,  kusama uses `9291` and westend uses `9290`. So, they all will read parachain inherents.

Can you elaborate which other runtimes should we care about? Do you mean previous versions? Do you mean custom runtimes?

Reply:
these 3 networks/runtimes are relay chains for parachains.  I don't think these inherents are required for running a standalone chain or a parachain.

Reply:
tested this to be working on following chains

- dev
- dev-staking
- dev-v3substrate
- gssmr
- gssmr-staking
- gssmr-v3substrate

So, it looks like in case parachain inherents are not needed, they just get ignored